Consolidate timezone utils

dannyroberts opened the pull request:
Biggest thing in this PR is heavily refactoring the way `adjust_datetime_to_timezone` is used throughout, to be more visually easy to see correctness. Also spotted what I believe to be a number of live timezone bugs, which I've left code comments about.

Going to just put this on staging for a while as I continue other tz work

dannyroberts committed:
move get_timezone => get_timezone_for_user

move corehq.apps.reports.util.get_timezone to
corehq.util.timezones.utils.get_timezone_for_user

dannyroberts committed:
move random pact thing out of dimagi utils

dannyroberts committed:
intorduce TIMEZONE_DATA_MIGRATION_COMPLETE placeholder

right now it is just set to False,
but I want to use it as a placehold for everywhere that the logic will change
after the migration

dannyroberts committed:
split calls to adjust_datetime_to_timezone out by function

and make timezone conversion between phone times and server times
contingent on TIMEZONE_DATA_MIGRATION_COMPLETE

dannyroberts committed:
abstrat tz adjustments with [Server|User|Phone]Time classes

dannyroberts committed:
be tighter about returning timezone niave datetimes when appropriate

dimagimon commented:
There are 13 errors:

* corehq/apps/reminders/forms.py, line 1799 - E261 at least two spaces before inline comment
* corehq/apps/reminders/views.py, line 152 - E203 whitespace before ':'
* corehq/apps/reminders/views.py, line 152 - E501 line too long (141 > 115 characters)
* corehq/apps/reminders/views.py, line 191 - E261 at least two spaces before inline comment
* corehq/apps/reminders/views.py, line 1306 - E203 whitespace before ':'
* corehq/apps/sms/views.py, line 815 - E203 whitespace before ':'
* corehq/util/timezones/conversions.py, line 47 - E501 line too long (116 > 115 characters)
* corehq/util/timezones/utils.py, line 9 - F401 'notify_exception' imported but unused
* corehq/util/timezones/utils.py, line 29 - F821 undefined name '_adjust_datetime_to_timezone'
* custom/_legacy/a5288/reports.py, line 81 - E501 line too long (117 > 115 characters)
* custom/fri/api.py, line 7 - F401 'ServerTime' imported but unused
* custom/fri/api.py, line 108 - E261 at least two spaces before inline comment
* custom/fri/reports/reports.py, line 389 - E501 line too long (117 > 115 characters)

snopoke commented on corehq/apps/adm/models.py:
```diff
@@ -348,8 +349,10 @@ def calculate_totals(self, column_data):
+            now = UserTime(
+                getattr(datespan, self.start_or_end or "enddate"),
+                datespan.timezone
+            ).server_time().done().replace(tzinfo=pytz.utc)
```
Seems weird to have `done()` in the middle of a chained method call. Is this a special case?

dannyroberts replied:
Yeah this `done()` thing is something I have mixed feelings about. The original idea is that the these *Time helper classes converted only between each other, and then to get a datetime back out, you have to explicitly call `.done()`. (So in this case, you call `.done()` to get a datetime and then call `datetime.replace` on it. (This code uses timezones sloppily; I believe this is here because the output of `dateutil.parse` was timezone aware and the author of it just wanted these things to be comparable to each other, which was kind of par for the course of how we were treating timezones 3 years ago.)

Anyway, this line is "right" but I'm not crazy about needing to call done on every function, so I'll probably change this api a bit in the near future.

snopoke commented on corehq/apps/reports/standard/monitoring.py:
```diff
@@ -187,7 +188,17 @@ def milestone(self):
+        # timezone stripping problem, and so I hadn't factored that in.
+        # As a result, this was two timezones off from correct
+        # and no one noticed all these years...
+        return datetime.datetime.utcnow()
```
wow!

czue replied:
hahaha, that's amazing

snopoke commented:
This is nice Danny. It makes it clear what is going on.

czue commented:
yeah on quick pass this looks like a huge improvement!

dannyroberts committed:
fix _adjust_utc_datetime_to_timezone and _adjust_phone_datetime_to_utc

dimagimon commented:
There are 13 errors:

* corehq/apps/reminders/forms.py, line 1799 - E261 at least two spaces before inline comment
* corehq/apps/reminders/views.py, line 152 - E203 whitespace before ':'
* corehq/apps/reminders/views.py, line 152 - E501 line too long (141 > 115 characters)
* corehq/apps/reminders/views.py, line 191 - E261 at least two spaces before inline comment
* corehq/apps/reminders/views.py, line 1306 - E203 whitespace before ':'
* corehq/apps/sms/views.py, line 815 - E203 whitespace before ':'
* corehq/util/timezones/conversions.py, line 47 - E501 line too long (116 > 115 characters)
* corehq/util/timezones/utils.py, line 9 - F401 'notify_exception' imported but unused
* corehq/util/timezones/utils.py, line 29 - F821 undefined name '_adjust_datetime_to_timezone'
* custom/_legacy/a5288/reports.py, line 81 - E501 line too long (117 > 115 characters)
* custom/fri/api.py, line 7 - F401 'ServerTime' imported but unused
* custom/fri/api.py, line 108 - E261 at least two spaces before inline comment
* custom/fri/reports/reports.py, line 389 - E501 line too long (117 > 115 characters)

dannyroberts commented:
FYI decided to pull this little guy out so it can get merged separately with it's corresponding dimagi utils change: https://github.com/dimagi/commcare-hq/pull/6038 and https://github.com/dimagi/dimagi-utils/pull/255

dannyroberts committed:
fix tz 500 in loading builds on release page

- make relase pages not use utc_to_timezone (it's a template tag anyway)
  in favor of something more tight and explicit
- make utc_to_timezone reject string dates as input, now that the only
  calls made that passed in strings are removed
- have [Server|User|Phone]Times assert things about their input to fail
  fast, have bugs come up early, and make debugging easier

dimagimon commented:
There are 13 errors:

* corehq/apps/reminders/forms.py, line 1799 - E261 at least two spaces before inline comment
* corehq/apps/reminders/views.py, line 152 - E203 whitespace before ':'
* corehq/apps/reminders/views.py, line 152 - E501 line too long (141 > 115 characters)
* corehq/apps/reminders/views.py, line 191 - E261 at least two spaces before inline comment
* corehq/apps/reminders/views.py, line 1306 - E203 whitespace before ':'
* corehq/apps/sms/views.py, line 815 - E203 whitespace before ':'
* corehq/util/timezones/conversions.py, line 47 - E501 line too long (116 > 115 characters)
* corehq/util/timezones/utils.py, line 9 - F401 'notify_exception' imported but unused
* corehq/util/timezones/utils.py, line 29 - F821 undefined name '_adjust_datetime_to_timezone'
* custom/_legacy/a5288/reports.py, line 81 - E501 line too long (117 > 115 characters)
* custom/fri/api.py, line 7 - F401 'ServerTime' imported but unused
* custom/fri/api.py, line 108 - E261 at least two spaces before inline comment
* custom/fri/reports/reports.py, line 389 - E501 line too long (117 > 115 characters)

dannyroberts committed:
add tests for timezone conversions

dimagimon commented:
There are 13 errors:

* corehq/apps/reminders/forms.py, line 1799 - E261 at least two spaces before inline comment
* corehq/apps/reminders/views.py, line 152 - E203 whitespace before ':'
* corehq/apps/reminders/views.py, line 152 - E501 line too long (141 > 115 characters)
* corehq/apps/reminders/views.py, line 191 - E261 at least two spaces before inline comment
* corehq/apps/reminders/views.py, line 1306 - E203 whitespace before ':'
* corehq/apps/sms/views.py, line 815 - E203 whitespace before ':'
* corehq/util/timezones/conversions.py, line 47 - E501 line too long (116 > 115 characters)
* corehq/util/timezones/utils.py, line 9 - F401 'notify_exception' imported but unused
* corehq/util/timezones/utils.py, line 29 - F821 undefined name '_adjust_datetime_to_timezone'
* custom/_legacy/a5288/reports.py, line 81 - E501 line too long (117 > 115 characters)
* custom/fri/api.py, line 7 - F401 'ServerTime' imported but unused
* custom/fri/api.py, line 108 - E261 at least two spaces before inline comment
* custom/fri/reports/reports.py, line 389 - E501 line too long (117 > 115 characters)

dannyroberts commented:
@czue @christineawhite This touches basically everywhere that we convert timezones, which means there's some risk of introducing a bug, so I want to be deliberate about how we treat this. So far here's what I've done to insure this doesn't cause bugs:
- was very meticulous about thinking through each case I was changing, detecting the intent behind the timezone conversion, making sure it was correct, fixing a couple that were incorrect, seeing where else the value was then used etc to check my logic
- my version of the code is stricter than the old functions (which would just kind of accept anything and try to make it work) and so I've added asserts to enforce assumptions I've made. This type of error will be super obvious to fix once it errors, so what I did was throw it up on staging and open every standard report to expose the most obvious ones. This exposed a few errors, which I fixed (see the commits that have comments with the ticked checkboxes).
- I added tests for the new conversion module's public interface (the ServerTime, etc. classes and their methods) to make sure I wasn't making some gross error in the actual timezone conversion. (I wasn't!)

If I deploy this right now, there are two types of errors that could happen:
- Some code 500s because it was abusing (ok fine, using) the previous conversion function's leniency. These will be relatively harmless in that they are super easy to fix, and will at most just cause a report to be unavailable for the time between when someone first tries it and when someone notices and fixes it. I'd give it a relatively high (25-50%?) that this happens somewhere, in some custom report or ADM feature, just because of how hard it is for me to test each of these cases individually
- The worse type of error would be that I wasn't meticulous enough and either made a typo or misunderstood the intention behind some conversion and didn't translate it correctly. It's hard for me to estimate the chance that I introduced such an error. I'd say low, but I'm biased by having been the one who did it.

My personal opinion is that the right thing to do here is to deploy this code now that it has no known bugs, and keep a keen eye out for any reports of incorrect dates in a report.

dannyroberts committed:
pep8

dimagimon commented:
There are 12 errors:

* corehq/apps/reminders/forms.py, line 1799 - E261 at least two spaces before inline comment
* corehq/apps/reminders/views.py, line 152 - E203 whitespace before ':'
* corehq/apps/reminders/views.py, line 152 - E501 line too long (141 > 115 characters)
* corehq/apps/reminders/views.py, line 191 - E261 at least two spaces before inline comment
* corehq/apps/reminders/views.py, line 1306 - E203 whitespace before ':'
* corehq/apps/sms/views.py, line 815 - E203 whitespace before ':'
* corehq/util/timezones/utils.py, line 9 - F401 'notify_exception' imported but unused
* corehq/util/timezones/utils.py, line 29 - F821 undefined name '_adjust_datetime_to_timezone'
* custom/_legacy/a5288/reports.py, line 81 - E501 line too long (117 > 115 characters)
* custom/fri/api.py, line 7 - F401 'ServerTime' imported but unused
* custom/fri/api.py, line 108 - E261 at least two spaces before inline comment
* custom/fri/reports/reports.py, line 389 - E501 line too long (117 > 115 characters)

snopoke commented:
We could also get a second person to go over each change and verify that
you've interpreted the original intention correctly.

On 26 March 2015 at 23:57, dimagimon <notifications@github.com> wrote:

> There are 12 errors:
>
>    - corehq/apps/reminders/forms.py, line 1799 - E261 at least two spaces
>    before inline comment
>    - corehq/apps/reminders/views.py, line 152 - E203 whitespace before ':'
>    - corehq/apps/reminders/views.py, line 152 - E501 line too long (141 >
>    115 characters)
>    - corehq/apps/reminders/views.py, line 191 - E261 at least two spaces
>    before inline comment
>    - corehq/apps/reminders/views.py, line 1306 - E203 whitespace before
>    ':'
>    - corehq/apps/sms/views.py, line 815 - E203 whitespace before ':'
>    - corehq/util/timezones/utils.py, line 9 - F401 'notify_exception'
>    imported but unused
>    - corehq/util/timezones/utils.py, line 29 - F821 undefined name
>    '_adjust_datetime_to_timezone'
>    - custom/_legacy/a5288/reports.py, line 81 - E501 line too long (117 >
>    115 characters)
>    - custom/fri/api.py, line 7 - F401 'ServerTime' imported but unused
>    - custom/fri/api.py, line 108 - E261 at least two spaces before inline
>    comment
>    - custom/fri/reports/reports.py, line 389 - E501 line too long (117 >
>    115 characters)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dimagi/commcare-hq/pull/6035#issuecomment-86732307>.
>



-- 
Simon Kelly
Senior Engineer | Dimagi South Africa

czue commented:
@dannyroberts i'm comfortable with your proposed approach, though agree that a thorough code review  is also a good idea. this PR is no longer mergeable fyi.

dannyroberts committed:
Merge branch 'master' into consolidate-timezone-utils

Conflicts:
	corehq/apps/app_manager/views.py

dimagimon commented:
There are 12 errors:

* corehq/apps/reminders/forms.py, line 1799 - E261 at least two spaces before inline comment
* corehq/apps/reminders/views.py, line 152 - E203 whitespace before ':'
* corehq/apps/reminders/views.py, line 152 - E501 line too long (141 > 115 characters)
* corehq/apps/reminders/views.py, line 191 - E261 at least two spaces before inline comment
* corehq/apps/reminders/views.py, line 1306 - E203 whitespace before ':'
* corehq/apps/sms/views.py, line 815 - E203 whitespace before ':'
* corehq/util/timezones/utils.py, line 9 - F401 'notify_exception' imported but unused
* corehq/util/timezones/utils.py, line 29 - F821 undefined name '_adjust_datetime_to_timezone'
* custom/_legacy/a5288/reports.py, line 81 - E501 line too long (117 > 115 characters)
* custom/fri/api.py, line 7 - F401 'ServerTime' imported but unused
* custom/fri/api.py, line 108 - E261 at least two spaces before inline comment
* custom/fri/reports/reports.py, line 389 - E501 line too long (117 > 115 characters)

dannyroberts commented:
Agree it'd be good to get another set of eyes on each change. Anyone want to volunteer? It may require opening this branch in your own editor so you can reason with the code more easily

dannyroberts committed:
Merge branch 'master' into consolidate-timezone-utils

Conflicts:
	corehq/ex-submodules/casexml/apps/case/models.py

dimagimon commented:
There are 12 errors:

* corehq/apps/reminders/forms.py, line 1799 - E261 at least two spaces before inline comment
* corehq/apps/reminders/views.py, line 152 - E203 whitespace before ':'
* corehq/apps/reminders/views.py, line 152 - E501 line too long (141 > 115 characters)
* corehq/apps/reminders/views.py, line 191 - E261 at least two spaces before inline comment
* corehq/apps/reminders/views.py, line 1306 - E203 whitespace before ':'
* corehq/apps/sms/views.py, line 815 - E203 whitespace before ':'
* corehq/util/timezones/utils.py, line 9 - F401 'notify_exception' imported but unused
* corehq/util/timezones/utils.py, line 29 - F821 undefined name '_adjust_datetime_to_timezone'
* custom/_legacy/a5288/reports.py, line 81 - E501 line too long (117 > 115 characters)
* custom/fri/api.py, line 7 - F401 'ServerTime' imported but unused
* custom/fri/api.py, line 108 - E261 at least two spaces before inline comment
* custom/fri/reports/reports.py, line 389 - E501 line too long (117 > 115 characters)

christineawhite commented:
@dannyroberts I'm happy with code review and then the wait and watch strategy

dannyroberts committed:
standardize datetime formatting in UI

- remove string_to_pretty_time
- add ui_string as replacement for calling strftime directly
- pull out safe_strftime for dates pre 1900, and use that in ui_string
- add USER_[DATE|TIME|DATETIME]_FORMAT constants to standardize output format
- formats include a timezone, which is a minor **user-facing change**

dannyroberts committed:
a little more ui_string cleanup

dimagimon commented:
There are 11 errors:

* corehq/apps/reminders/forms.py, line 1799 - E261 at least two spaces before inline comment
* corehq/apps/reminders/views.py, line 152 - E203 whitespace before ':'
* corehq/apps/reminders/views.py, line 152 - E501 line too long (141 > 115 characters)
* corehq/apps/reminders/views.py, line 191 - E261 at least two spaces before inline comment
* corehq/apps/reminders/views.py, line 1315 - E203 whitespace before ':'
* corehq/ex-submodules/phonelog/reports.py, line 333 - E127 continuation line over-indented for visual indent
* custom/_legacy/a5288/reports.py, line 81 - E501 line too long (117 > 115 characters)
* custom/_legacy/hsph/reports/old/project_management.py, line 199 - E501 line too long (116 > 115 characters)
* custom/fri/api.py, line 7 - F401 'ServerTime' imported but unused
* custom/fri/api.py, line 108 - E261 at least two spaces before inline comment
* custom/fri/reports/reports.py, line 389 - E501 line too long (117 > 115 characters)

dannyroberts commented:
(Deleting all the dimagimon linter comments to make this PR easier to read. The PR touches 2-5 lines of a lot code, and it didn't feel appropriate to be making PEP8 changes in what might be a long-lived PR, potentially creating merge conflicts.)

snopoke commented:
I'll take a look at this on Monday if no one gets to it before then

dannyroberts commented:
Thanks @snopoke that'll be super helpful

snopoke commented on corehq/util/timezones/conversions.py:
```diff
@@ -0,0 +1,111 @@
+
+    """
+    assert value.tzinfo is None
+    return from_tz.localize(value).astimezone(pytz.utc).replace(tzinfo=None)
```
This returns a timezone-naive datetime but the doc string says it returns a timezone-aware datetime.

dannyroberts replied:
Thanks, fixed!

snopoke commented on corehq/apps/reports/standard/cases/basic.py:
```diff
@@ -321,10 +322,6 @@ def rows(self):
-            # return tz_utils.adjust_datetime_to_timezone(
-            #     date, pytz.utc.zone, self.timezone.zone
-            # ).strftime('%Y-%m-%d %H:%M:%S')
+            return PhoneTime(date, self.timezone).user_time(self.timezone).done().strftime('%Y-%m-%d %H:%M:%S')
```
This seems a bit roundabout but it does make the it clear what's going on.

dannyroberts replied:
Yeah the other benefit is that this code continues to work after flipping `TIMEZONE_DATA_MIGRATION_COMPLETE`, actually doing the timezone conversion once that's necessary.

snopoke commented on corehq/apps/reports/standard/monitoring.py:
```diff
@@ -267,9 +278,9 @@ def get_number_cases(self, user_id, modified_after=None, modified_before=None, c
+            kwargs['modified_on__gte'] = ServerTime(modified_after).phone_time(self.timezone).done()
         if modified_before:
-            kwargs['modified_on__lt'] = modified_before
+            kwargs['modified_on__lt'] = ServerTime(modified_before).phone_time(self.timezone).done()
```
ncie

snopoke commented on corehq/apps/reports/templatetags/timezone_tags.py:
```diff
@@ -1,20 +1,16 @@
-            return date
-    return tz_utils.adjust_datetime_to_timezone(date, pytz.utc, timezone.zone).strftime(dest_fmt)
+    if isinstance(date, basestring):
+        raise ValueError("utc_to_timezone no longer accepts strings")
```
:boom:

snopoke commented on corehq/ex-submodules/phonelog/reports.py:
```diff
@@ -328,9 +329,8 @@ def _create_rows(self, logs, matching_id=None):
         for log in logs.order_by(self.ordering)[paged]:
-            date = str(log.date)
-            date_fmt = tz_utils.string_to_pretty_time(
-                date, self.timezone, fmt="%b %d, %Y %H:%M:%S")
```
that's a bit odd: convert date to string -> parse date -> change timezone -> format

dannyroberts replied:
Yeah right? There were a few places during this little adventure where I saw stuff like this—one of the big motivations for making these functions stricter about what they accept and specifically less fast-and-loose with string-formatted dates as inputs.

snopoke replied:
:+1:

snopoke commented on corehq/util/dates.py:
```diff
@@ -18,3 +18,28 @@ def get_timestamp(date):
     return 1000 * get_timestamp(date)
+
+
+def safe_strftime(val, fmt):
```
I don't see the necessitiy of this

dannyroberts replied:
Oops this was supposed to have been moved from another file, but I messed up the commit.

The main point of it is to avoid the following error in an (albeit edge) case where you're passing in a pre-1900 date/time:

```python
In [1]: import datetime

In [2]: datetime.date(1899, 1, 1).strftime('%Y')
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-2-db7a1f50dcad> in <module>()
----> 1 datetime.date(1899, 1, 1).strftime('%Y')

ValueError: year=1899 is before 1900; the datetime strftime() methods require year >= 1900
```

dannyroberts replied:
That should really be part of the comment, I'll add it, and remove the original function.

dannyroberts replied:
(done)

snopoke replied:
Interesting, definitely and edge case. Could also just check the the year is pre 1900 and if not proceed normally.

snopoke commented on custom/apps/crs_reports/reports.py:
```diff
@@ -189,9 +190,7 @@ def base_filters(self):
-        return tz_utils.adjust_datetime_to_timezone\
-            (date, pytz.utc.zone, self.timezone.zone).strftime\
-            ('%d-%m-%Y') if date else ""
+        return ServerTime(date).user_time(self.timezone).done().strftime('%d-%m-%Y') if date else ""
```
I think this should be the same as here: https://github.com/dimagi/commcare-hq/pull/6035/files#diff-64271a8d193a39ddd8979432c1da0d21R323

The date is a date in the case e.g. 'date_birth', 'opened_on', 'modified_on', 'closed_on'

dannyroberts replied:
Ah good catch, I was assuming the original code was correct when I was first going through, and didn't make the connection that this suffered from the same bug that was fixed and commented on in the other `date_to_json` function. Fixing!

snopoke commented on custom/bihar/reports/mch_reports.py:
```diff
@@ -59,9 +59,7 @@ def rendered_report_title(self):
-        return tz_utils.adjust_datetime_to_timezone\
-            (date, pytz.utc.zone, self.timezone.zone).strftime\
-            ('%d/%m/%Y') if date else ""
+        return ServerTime(date).user_time(self.timezone).done().strftime('%d/%m/%Y') if date else ""
```
Same as other `date_to_json` functions

snopoke commented on custom/fri/api.py:
```diff
@@ -260,11 +261,9 @@ def get_num_missed_windows(case):
-    )
+    # this was wrong before I refactored it to these "typed" datetimes
+    # with them, it just popped out at me
+    opened_timestamp = PhoneTime(case.opened_on).user_time(domain_obj.default_timezone)
```
I think you need to call `done()`

snopoke replied:
:+1: on refactor

snopoke replied:
+1 on refactor

dannyroberts replied:
Thanks, done()

snopoke commented:
@dannyroberts finished my detailed review, see individual comments for specifics.

dannyroberts committed:
fix comments about timezone-awareness in conversions.py

dimagimon commented:
There are 11 errors:

* corehq/apps/reminders/forms.py, line 1799 - E261 at least two spaces before inline comment
* corehq/apps/reminders/views.py, line 152 - E203 whitespace before ':'
* corehq/apps/reminders/views.py, line 152 - E501 line too long (141 > 115 characters)
* corehq/apps/reminders/views.py, line 191 - E261 at least two spaces before inline comment
* corehq/apps/reminders/views.py, line 1315 - E203 whitespace before ':'
* corehq/ex-submodules/phonelog/reports.py, line 333 - E127 continuation line over-indented for visual indent
* custom/_legacy/a5288/reports.py, line 81 - E501 line too long (117 > 115 characters)
* custom/_legacy/hsph/reports/old/project_management.py, line 199 - E501 line too long (116 > 115 characters)
* custom/fri/api.py, line 7 - F401 'ServerTime' imported but unused
* custom/fri/api.py, line 108 - E261 at least two spaces before inline comment
* custom/fri/reports/reports.py, line 389 - E501 line too long (117 > 115 characters)

dannyroberts committed:
further safe_strftime cleanup and commenting

dimagimon commented:
There are 11 errors:

* corehq/apps/reminders/forms.py, line 1799 - E261 at least two spaces before inline comment
* corehq/apps/reminders/views.py, line 152 - E203 whitespace before ':'
* corehq/apps/reminders/views.py, line 152 - E501 line too long (141 > 115 characters)
* corehq/apps/reminders/views.py, line 191 - E261 at least two spaces before inline comment
* corehq/apps/reminders/views.py, line 1315 - E203 whitespace before ':'
* corehq/ex-submodules/phonelog/reports.py, line 333 - E127 continuation line over-indented for visual indent
* custom/_legacy/a5288/reports.py, line 81 - E501 line too long (117 > 115 characters)
* custom/_legacy/hsph/reports/old/project_management.py, line 199 - E501 line too long (116 > 115 characters)
* custom/fri/api.py, line 7 - F401 'ServerTime' imported but unused
* custom/fri/api.py, line 108 - E261 at least two spaces before inline comment
* custom/fri/reports/reports.py, line 389 - E501 line too long (117 > 115 characters)

dannyroberts committed:
fix date_to_json methods

dimagimon commented:
There are 11 errors:

* corehq/apps/reminders/forms.py, line 1799 - E261 at least two spaces before inline comment
* corehq/apps/reminders/views.py, line 152 - E203 whitespace before ':'
* corehq/apps/reminders/views.py, line 152 - E501 line too long (141 > 115 characters)
* corehq/apps/reminders/views.py, line 191 - E261 at least two spaces before inline comment
* corehq/apps/reminders/views.py, line 1315 - E203 whitespace before ':'
* corehq/ex-submodules/phonelog/reports.py, line 333 - E127 continuation line over-indented for visual indent
* custom/_legacy/a5288/reports.py, line 81 - E501 line too long (117 > 115 characters)
* custom/_legacy/hsph/reports/old/project_management.py, line 199 - E501 line too long (116 > 115 characters)
* custom/fri/api.py, line 7 - F401 'ServerTime' imported but unused
* custom/fri/api.py, line 108 - E261 at least two spaces before inline comment
* custom/fri/reports/reports.py, line 389 - E501 line too long (117 > 115 characters)

dannyroberts committed:
add missing .done() on tz conversion

dimagimon commented:
There are 11 errors:

* corehq/apps/reminders/forms.py, line 1799 - E261 at least two spaces before inline comment
* corehq/apps/reminders/views.py, line 152 - E203 whitespace before ':'
* corehq/apps/reminders/views.py, line 152 - E501 line too long (141 > 115 characters)
* corehq/apps/reminders/views.py, line 191 - E261 at least two spaces before inline comment
* corehq/apps/reminders/views.py, line 1315 - E203 whitespace before ':'
* corehq/ex-submodules/phonelog/reports.py, line 333 - E127 continuation line over-indented for visual indent
* custom/_legacy/a5288/reports.py, line 81 - E501 line too long (117 > 115 characters)
* custom/_legacy/hsph/reports/old/project_management.py, line 199 - E501 line too long (116 > 115 characters)
* custom/fri/api.py, line 7 - F401 'ServerTime' imported but unused
* custom/fri/api.py, line 108 - E261 at least two spaces before inline comment
* custom/fri/reports/reports.py, line 389 - E501 line too long (117 > 115 characters)

snopoke commented on corehq/util/timezones/conversions.py:
```diff
@@ -0,0 +1,113 @@
+from corehq.const import USER_DATETIME_FORMAT
+from corehq.util.dates import safe_strftime
+
+TIMEZONE_DATA_MIGRATION_COMPLETE = False
```
A brief comment here would be good just outlining what's expected to change. Not essential though.

dannyroberts replied:
Good idea. Added comment.

On Mon, Mar 30, 2015 at 11:38 AM, Simon Kelly <notifications@github.com>
wrote:

> In corehq/util/timezones/conversions.py
> <https://github.com/dimagi/commcare-hq/pull/6035#discussion_r27401430>:
>
> > @@ -0,0 +1,113 @@
> > +import pytz
> > +from corehq.const import USER_DATETIME_FORMAT
> > +from corehq.util.dates import safe_strftime
> > +
> > +TIMEZONE_DATA_MIGRATION_COMPLETE = False
>
> A brief comment here would be good just outlining what's expected to
> change. Not essential though.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dimagi/commcare-hq/pull/6035/files#r27401430>.
>

snopoke commented:
I think this is ready to merge. Just waiting for tests. Feel free to merge when they pass.

dannyroberts committed:
add comment to TIMEZONE_DATA_MIGRATION_COMPLETE

dimagimon commented:
There are 11 errors:

* corehq/apps/reminders/forms.py, line 1799 - E261 at least two spaces before inline comment
* corehq/apps/reminders/views.py, line 152 - E203 whitespace before ':'
* corehq/apps/reminders/views.py, line 152 - E501 line too long (141 > 115 characters)
* corehq/apps/reminders/views.py, line 191 - E261 at least two spaces before inline comment
* corehq/apps/reminders/views.py, line 1315 - E203 whitespace before ':'
* corehq/ex-submodules/phonelog/reports.py, line 333 - E127 continuation line over-indented for visual indent
* custom/_legacy/a5288/reports.py, line 81 - E501 line too long (117 > 115 characters)
* custom/_legacy/hsph/reports/old/project_management.py, line 199 - E501 line too long (116 > 115 characters)
* custom/fri/api.py, line 7 - F401 'ServerTime' imported but unused
* custom/fri/api.py, line 108 - E261 at least two spaces before inline comment
* custom/fri/reports/reports.py, line 389 - E501 line too long (117 > 115 characters)

dannyroberts commented:
Thanks @snopoke for the review and for finding some errors I hadn't caught

dannyroberts committed:
optimize timezone imports

[ci skip]

dimagimon commented:
There are 11 errors:

* corehq/apps/reminders/forms.py, line 1799 - E261 at least two spaces before inline comment
* corehq/apps/reminders/views.py, line 151 - E203 whitespace before ':'
* corehq/apps/reminders/views.py, line 151 - E501 line too long (141 > 115 characters)
* corehq/apps/reminders/views.py, line 190 - E261 at least two spaces before inline comment
* corehq/apps/reminders/views.py, line 1314 - E203 whitespace before ':'
* corehq/apps/reports/standard/sms.py, line 21 - F401 'mobile' imported but unused
* corehq/ex-submodules/phonelog/reports.py, line 332 - E127 continuation line over-indented for visual indent
* custom/_legacy/a5288/reports.py, line 78 - E501 line too long (117 > 115 characters)
* custom/_legacy/hsph/reports/old/project_management.py, line 199 - E501 line too long (116 > 115 characters)
* custom/fri/api.py, line 106 - E261 at least two spaces before inline comment
* custom/fri/reports/reports.py, line 387 - E501 line too long (117 > 115 characters)

dannyroberts commented:
Ok merging since the last non-comment, non-import-cleanup change's build passed. Last change (https://github.com/dimagi/commcare-hq/commit/a6e50aa023e52c0a91e267773c0667cd5e15d5df) is just removing imports that this PR got rid of (and other unused imports nearby them).

czue commented:
:clap:
